Detectors/TPC/simulation/test/testTPCDigitContainer.cxx

```diff
@@ -73,48 +72,51 @@ namespace TPC {
   /// and that the MC labels are right
   BOOST_AUTO_TEST_CASE(DigitContainer_test2)
   {
+    const Mapper& mapper = Mapper::instance();
     const SAMPAProcessing& sampa = SAMPAProcessing::instance();
+    static FairRootManager *mgr = FairRootManager::Instance();
     auto *mDigitContainer = new DigitContainer();
 
-    /// \todo add test with FairLink stuff if possible
-    std::vector<int> hitID = { 1, 62,  1, 62, 62, 50, 62, 1, 1, 1};
-    //std::vector<int> MCevent = { 1, 62,  1, 62, 62, 50, 62, 1, 1, 1};
-    //std::vector<int> MCtrack = {22,  3, 22,  3,  3, 70,  3, 7, 7, 7};
+    std::vector<int> MCevent = { 1, 62,  1, 62, 62, 50, 62, 1, 1, 1};
+    std::vector<int> MCtrack = {22, 3, 22, 3, 3, 70, 3, 7, 7, 7};
     std::vector<int> CRU     = {23, 23, 23, 23, 23, 23, 23, 23, 23, 23};
     std::vector<int> Time    = {231, 231, 231, 231, 231, 231, 231, 231, 231, 231};
     std::vector<int> Row     = {12, 12, 12, 12, 12, 12, 12, 12, 12, 12};
     std::vector<int> Pad     = {15, 15, 15, 15, 15, 15, 15, 15, 15, 15};
-    std::vector<int> nEle    = {60, 1, 252, 10, 2, 3, 5, 25, 24, 23};
+    std::vector<float> nEle  = {60, 1, 252, 10, 2, 3, 5, 25, 24, 23};
 
     std::vector<int> MCeventSorted = {62, 1, 1, 50};
     std::vector<int> MCtrackSorted = {3, 7, 22, 70};
 
-    int nEleSum = 0;
+    float nEleSum = 0;
     for(int i=0; i<CRU.size(); ++i) {
-      //mDigitContainer->addDigit(MCevent[i], MCtrack[i], CRU[i], Time[i], Row[i], Pad[i], nEle[i]);
-      mDigitContainer->addDigit(hitID[i], CRU[i], Time[i], Row[i], Pad[i], nEle[i]);
+      mgr->SetEntryNr(MCevent[i]);
+      mDigitContainer->addDigit(MCtrack[i], CRU[i], Time[i], Row[i], Pad[i], nEle[i]);
       nEleSum += nEle[i];
     }
 
     auto *mDigitsArray = new TClonesArray("o2::TPC::DigitMC");
```

Comment:
here it does not work. In the next line, the function fillOutputContainer requests a pointer, which is owed to the internal handling of the TClonesArrays in FairRoot (usually the mDigitsArray is what is registered to the FairRootManager)